google-cloud-logging/src/main/java/com/google/cloud/logging/LoggingImpl.java

```diff
@@ -443,11 +444,21 @@ public boolean deleteLog(String log) {
     return get(deleteLogAsync(log));
   }
 
+  public boolean deleteLog(String log, LogDestinationName destination) {
+    return get(deleteLogAsyncImpl(log, destination));
+  }
+
   public ApiFuture<Boolean> deleteLogAsync(String log) {
-    DeleteLogRequest request =
-        DeleteLogRequest.newBuilder()
-            .setLogName(LogName.ofProjectLogName(getOptions().getProjectId(), log).toString())
-            .build();
+    return deleteLogAsyncImpl(log, null);
+  }
+
+  public ApiFuture<Boolean> deleteLogAsync(String log, LogDestinationName destination) {
+    return deleteLogAsyncImpl(log, destination);
+  }
+
+  private ApiFuture<Boolean> deleteLogAsyncImpl(String log, LogDestinationName destination) {
+    LogName name = getLogName(getOptions().getProjectId(), log, destination, true);
```

Comment:
nit: mind that passing `true` you modify the existing behavior of the `deleteLog*(String)` method. I am OK with either option: throwing NPE or failing at backend, but it should be unanimous for all methods (not only `deleteLog*`).

Reply:
Nope, `LogName` constructor is using `Preconditions.checkNotNull` and previous code was calling the `LogName.ofProjectLogName` which constructs `LogName`